worldwind/src/commonMain/kotlin/earth/worldwind/shape/Ellipse.kt

```diff
@@ -441,15 +507,113 @@ open class Ellipse @JvmOverloads constructor(
             boundingSector.setEmpty()
         }
     }
+    protected open fun addLineVertex(
+        rc: RenderContext, latitude: Angle, longitude: Angle, altitude: Double, offset : Int, firstOrLast : Boolean
+    )
+    {
+        val vertex = (lineVertexIndex / LINE_VERTEX_STRIDE - 1) * 2
+        var point = rc.geographicToCartesian(latitude, longitude, altitude, altitudeMode, scratchPoint)
+        if (lineVertexIndex == 0) texCoord1d = 0.0
+        else texCoord1d += point.distanceTo(prevPoint)
+        prevPoint.copy(point)
+        if (isSurfaceShape) {
+            lineVertexArray[lineVertexIndex++] = (longitude.inDegrees - vertexOrigin.x).toFloat()
+            lineVertexArray[lineVertexIndex++] = (latitude.inDegrees - vertexOrigin.y).toFloat()
+            lineVertexArray[lineVertexIndex++] = (altitude - vertexOrigin.z).toFloat()
+            lineVertexArray[lineVertexIndex++] = 1.0f
```

Comment:
Nitpicking unless you use w no need to waste memory on constant

Reply:
it is used to determine in which direction this vertex should be moved for lines